### ISSUE TITLE
litServe

### ISSUE THREAD

**Issue by Borda** [pytorch-fdn/ecosystem#27]
### Contact emails

jirka@lightning.ai

### Project summary

Lightning-fast serving engine for any AI model of any size. Flexible. Easy. Enterprise-scale.

### Project description

LitServe is an easy-to-use, flexible serving engine for AI models built on FastAPI. It augments FastAPI with features like batching, streaming, and GPU autoscaling eliminate the need to rebuild a FastAPI server per model.

### Are there any other projects in the PyTorch Ecosystem similar to yours?  If, yes, what are they?

FastAPI
TorchServe

### Project repo URL

https://github.com/Lightning-AI/LitServe

### Project license

https://github.com/Lightning-AI/LitServe/blob/main/LICENSE

### GitHub handles of the project maintainer(s)

@aniketmaurya, @williamFalcon, @lantiga, @ali-alshaar7, @borda

### Is there a corporate or academic entity backing this project?  If so, please provide the name and URL of the entity.

Lightning AI, https://lightning.ai/

### Website URL

https://lightning.ai/docs/litserve/home

### Documentation

https://lightning.ai/docs/litserve/home

### How do you build and test the project today (continuous integration)?  Please describe.

full testing - https://github.com/Lightning-AI/LitServe/actions/workflows/ci-testing.yml
including multi-GPU - https://dev.azure.com/Lightning-AI/lightning/_build?definitionId=75

### Version of PyTorch

`torch >2.0.0`

### Components of PyTorch

```py
torch.stack
torch.backends
```

### How long do you expect to maintain the project?

indefinitely

**Comment by Borda** [comment]
@michellearoth, just curious, why did you reopen this issue?

**Comment by michellearoth** [comment]
@EmreGuven-0 can you please do the 6 month review for litServe?

**Comment by michellearoth** [comment]
@EmreGuven-0 reminder for the 6 month review - let us know if we need to assign someone else.

**Comment by EmreGuven-0** [comment]
6 month review results below, using Claude and then verifying with Codex. See below.

## 6-Month Ecosystem Review: LitServe - Jun 24, 2026

### Functional Requirements
- ✅ **PyTorch Support:** Serves any PyTorch model. Uses `torch.stack` (batching) and `torch.backends` (device detection), imported lazily; torch is an optional/dev dependency. Validated against current torch on CPU (CI matrix) and GPU (`.lightning` L4 pipeline).
- ✅ **Working CI:** Five GitHub Actions workflows (ci-testing on ubuntu/macos/windows × Py3.10–3.13 + oldest-deps + daily cron; ci-checks; FastAPI parity; minimal-dependency check) plus a multi-GPU pipeline in `.lightning` (L4_X_2, asserts ≥2 GPUs) and Codecov.
- ⚠️ **Governance:** No governance doc in the repo. `CODEOWNERS` exists but does not document the maintainer model / decision process. Recommendation: add a short `GOVERNANCE.md` (LF Minimum Viable Governance style).
- ✅ **User experience:** README states the value proposition clearly (custom inference servers in pure Python; control over batching, streaming, scaling).
- ✅ **Quick start:** `pip install litserve` + minimal server example + `curl` test in the README, plus a docs "Get started" path.
- ✅ **Documentation:** Full docs site (lightning.ai/docs/litserve) with features, API reference, examples, benchmarks; 100+ templates.
- ✅ **Permissive licensing:** Apache-2.0.
- ✅ **Functional Testing:** Extensive suite (unit / integration / e2e / parity / perf), pytest with strict markers, Codecov targets 99% project / 50% patch.
- ✅ **Packages:** `pip install litserve` on PyPI, automated via `release-pypi.yml`.
- ⚠️ **Ongoing maintenance:** Maintenance commitment stated as "indefinitely" and the repo is actively pushed (last push 23 Jun 2026), but release cadence has paused since Dec 2025 and commit volume is low.

### Measurable Requirements
- ✅ **Core Maintainers (≥2):** `CODEOWNERS` lists 10 owners (lantiga, aniketmaurya, ethanwharris, justusschock, tchaton, andyland, k223kim, KaelanDt, dmitsf, Abdul-0x4A).
- ⚠️ **Contributors active in last 90 days (≥5):** Only 4 distinct authors touched `main` since 26 Mar 2026, of which 2 are human (@bhimrazy, @Lidang-Jiang) and 2 are bots (dependabot, pre-commit-ci). Below the minimum.
- ⚠️ **Commits in last 90 days (≥30):** 12 commits to `main` since 26 Mar 2026 (7 excluding dependabot/pre-commit bots). Well below the minimum, even allowing the usual grace on this metric.
- ✅ **Stars (≥200):** 3,898 stars.

### Summary
LitServe remains a mature, high-value, permissively licensed project with excellent CI (multi-OS and multi-GPU), a deep test suite, strong docs and quick start, automated PyPI releases, and an experienced maintainer group backed by Lightning AI. It is actively maintained (last push 23 Jun 2026).

However, as of this review it does **not meet two measurable ecosystem thresholds**: trailing-90-day commits (12 vs ≥30) and active contributors (2 human / 4 total vs ≥5), and it is **missing a repo file** the checklist expects (GOVERNANCE.md).

**Comment by michellearoth** [comment]
@aniketmaurya, @williamFalcon, @lantiga, @ali-alshaar7, @Borda can you please respond to the above concerns?

**Comment by bhimrazy** [comment]
Hi @EmreGuven-0,

Thanks for the detailed 6-month review, and apologies for the delayed response. Here's where things stand on the points you raised:

1. **Governance**: added [`GOVERNANCE.md`](https://github.com/Lightning-AI/LitServe/blob/main/GOVERNANCE.md). It covers roles, adding and retiring maintainers, decision making, and releases. We also added a [`CODE_OF_CONDUCT.md`](https://github.com/Lightning-AI/LitServe/blob/main/CODE_OF_CONDUCT.md) and a [`CONTRIBUTING.md`](https://github.com/Lightning-AI/LitServe/blob/main/CONTRIBUTING.md).
2. **Maintainers**: [`.github/CODEOWNERS`](https://github.com/Lightning-AI/LitServe/blob/main/.github/CODEOWNERS) now points to the governance doc and lists current maintainers separately from retired ones. It's the up-to-date list, since some of the handles in the original application have since moved on.
3. **Releases**: releases have resumed: [v0.2.18](https://github.com/Lightning-AI/LitServe/releases/tag/v0.2.18) (Aug 12) and [v0.2.19](https://github.com/Lightning-AI/LitServe/releases/tag/v0.2.19) (Sep 9).
4. **Activity**: we're aware it's been below the thresholds. It's picking up: more contributors are active and several community PRs are close to merging, so we expect it to keep improving.

Regards,
Bhimraj Yadav, on behalf of the LitServe maintainers

cc: @andyland @dmitsf @tchaton

fyi: @michellearoth
